bin/pygrb/pycbc_grb_inj_finder

```diff
@@ -95,56 +93,101 @@ def read_hdf5_triggers(inputfiles, verbose=False):
+                "Cannot merge {0}/{1}, does not match dtype".format(
+                    obj.file.filename, name,
+                ))
+        datasets_n[name] = (shape, dtype)
 
     # get list of datasets
-    datasets = {}
+    datasets_n = {}
+    datasets_ifo = {}
     for filename in inputfiles:
         with h5py.File(filename, 'r') as h5f:
-            h5f.visititems(_scan_dataset)
+            h5f.visititems(_scan_dataset_ifo)
+            h5f.visititems(_scan_dataset_network)
 
     position = defaultdict(int)
-
-    out = {}
-
-    # create datasets
-    for dset, (shape, dtype) in datasets.items():
-        # 8: chops off network/ from the dset string
-        out[dset[8:]] = numpy.empty(shape, dtype=dtype)
+    
+    out_dict = {}
+    for x in trig_dict:
+        out_dict[x] = {}
+        
+    placeholder = 0
+
+    # create datasets    
+    for dset, (shape, dtype) in datasets_ifo.items():
+        # 3: chops off H1/ or L1/ or V1/ from the dset string
+        for x in trig_dict:
+            if placeholder == 0 and x!= 'network':
+                placeholder = x
+            if dset.startswith(x):
+                out_dict[x][dset[3:]] = numpy.empty(shape, dtype=dtype)
+            
+    for dset, (shape, dtype) in datasets_n.items():
+        out_dict['network'][dset[8:]] = numpy.empty(shape, dtype=dtype)
 
     # copy dataset contents
     for filename in inputfiles:
         with h5py.File(filename, 'r') as h5in:
-            for dset in datasets:
-                data = h5in[dset][:]
-                size = data.shape[0]
-                pos = position[dset]
-                if EVENT_ID_REGEX.search(dset):
-                    out[dset[8:]][pos:pos+size] = data + pos
-                else:
-                    out[dset[8:]][pos:pos+size] = data
-                position[dset] += size
-
-    return out
-
+            if len(h5in[placeholder].keys()) != 1:
+                for dset in datasets_ifo:            
+                    for x in trig_dict:
+                        if dset[:2] == x and x!='network':
+                            data = h5in[dset][:]
+                            size = data.shape[0]
+                            pos = position[dset]
+                            if EVENT_ID_REGEX.search(dset):
+                                out_dict[x][dset[3:]][pos:pos+size] = data + pos
+                            else:
+                                out_dict[x][dset[3:]][pos:pos+size] = data
+                            position[dset] += size
+                        
+                for dset in datasets_n: 
+                    data = h5in[dset][:]
+                    size = data.shape[0]
+                    pos = position[dset]
+                    if EVENT_ID_REGEX.search(dset):
+                        out_dict['network'][dset[8:]][pos:pos+size] = data + pos
+                    else:
+                        out_dict['network'][dset[8:]][pos:pos+size] = data
+                    position[dset] += size
```

Comment:
It should be possible to turn these 2 for loops into 2 calls to a single function (with appropriate arguments)